IanGRap/Squish
# Lobby team hat refactor

Contributors: Alfredo

This pull request tries rework how teams are identified. Hats will now be used to represent a team. Skins can be chosen similar to FreeForAll modes

Changes: 
* PlayerHatsScriptableObject.cs: a new script for a ScriptaleObject that holds the list of hats. I created this to try to separate the hats list from PlayerHatChanger.cs. It seemed like they shouldn't be bound together
* Teams.cs: added variables to hold the info for each teams hat
* RegisterTeamMembers.cs: jumping on the podiums should give the player a hat
* ControllerSignIn.cs: lets ppl switch skins in the team lobby
* PlayerHatChanger.cs: removed the _hatsInUse variable. After looking through the script it seemed to me that this variable wasn't doing anything useful. SetHat() is now public
* PlayerSelectUiManager.cs: Ready/Start Ui no longer gives the icons a color based on teams
* Materials.cs: no longer use the team skins list so SwapList() isn't needed

How to Test: 
* run through the Team Lobby
* spawn in. You should spawn in without a hat.
* When you jump on a podium it will give you that teams hat (horns or halo)
* Also go through the regular FreeForAll lobby to make sure I didn't break that hat system

Note: As mentioned above, I created a scriptable object to hold the list of hats. Before, this list was held in the PlayerHatChanger.cs. I didn't want for a script to need PlayerHatChanger if it needed that list of hats. However, the one script that I did add this scriptable object to also needed the PlayerHatChanger script. So i'm not sure if I should revert the change and put the list back into the PlayerHatChanger.cs script.
